base/special/trig.jl

```diff
@@ -106,7 +106,7 @@ Compute ``\\sin(\\pi x)`` more accurately than `sin(pi*x)`, especially for large
 function sinpi(x::T) where T<:AbstractFloat
     if !isfinite(x)
         isnan(x) && return x
-        throw(DomainError())
+        throw(DomainError(x, "`x` cannot be infinite."))
```

Comment:
I believe that's intentional, as some of the methods check for `isnan` and return the input, others don't